mongo/single_result.go

```diff
@@ -28,6 +29,22 @@ type SingleResult struct {
 	reg *bsoncodec.Registry
 }
 
+// NewSingleResultFromBytes creates a SingleResult with an underlying Cursor pre-loaded with the provided contents.
+func NewSingleResultFromBytes(contents []byte) *SingleResult {
+	c := &Cursor{
+		bc:       driver.NewBatchCursorFromBytes(contents),
+		registry: bson.DefaultRegistry,
```

Comment:
We could let users specify a custom `registry` here.

Reply:
Changed `NewMockSingleResult` and `NewMockCursor` to take in a `bsoncodec.Registry`.